drivers/soundwire/stream.c

```diff
@@ -1843,29 +1843,27 @@ int sdw_stream_add_master(struct sdw_bus *bus,
 		ret = -ENOMEM;
 		goto unlock;
 	}
+skip_alloc_master_rt:
+
+	ret = sdw_master_port_alloc(m_rt, num_ports);
+	if (ret)
+		goto alloc_error;
+
+	stream->m_rt_count++;
 
 	ret = sdw_master_rt_config(m_rt, stream_config);
 	if (ret < 0)
 		goto unlock;
 
-skip_alloc_master_rt:
 	ret = sdw_config_stream(bus->dev, stream, stream_config, false);
 	if (ret)
-		goto stream_error;
-
-	ret = sdw_master_port_alloc(m_rt, num_ports);
-	if (ret)
-		goto stream_error;
+		goto unlock;
```

Comment:
Shouldn't this be `goto alloc_error`? And line 1856 too.

Reply:
actually no @bardliao, it's intentional, see commit message:

````
This is an invasive patch which modifies the error handling flow, with
cleanups only done when an allocation fails. Configuration failures
only return an error code.
````

Reply:
@plbossart Ok, I may still miss something, and didn't get why we only want to free `m_rt` when `sdw_master_port_alloc` failed, yet. Or the question should be why do we need to free `m_rt` when `sdw_master_port_alloc` failed? The `m_rt` may not even allocated by `sdw_stream_add_master`

Reply:
Good question @bardliao. I didn't really change this code, this was the same before.
But I agree that I could add a test to only free m_rt if it's allocated here.